app/Models/User.php

```diff
@@ -540,7 +540,7 @@ public function rank()
 
     public function rankHistories()
     {
-        return $this->hasMany(RankHistory::class);
+        return $this->hasMany("App\Models\RankHistory", 'user_id', 'user_id');
```

Comment:
the statement was unlike all the others, and the 'user_id' column needed to be defined for it to work.